### ISSUE TITLE
Interactive and recursive grep/ack

### ISSUE THREAD

**Issue by amelio-vazquez-reina** [emacs-helm/helm#573]
I love helm. It's an **incredibly** useful Emacs package, period, and I want to thank @thierryvolpiatto and the rest of the helm dev team for their hard work and dedication to making this package as great as it is.

Now, I have a question about `helm-ack` and `helm-do-grep`. With `helm-do-grep`, I can interactively search within files in one folder (that is, as I type, `helm-do-grep` shows the results of the search, which is extremely useful). However, it only goes one level deep. Is it possible to change this?

With `helm-ack`, I can `ack` recursively in a folder (i.e. multiple levels deep), but it is not interactive, i.e. I need to type: `keyword RET` in order to see the result. This takes me to the following question:
1. Is there a way to have `helm-ack` show results interactively?
2. Alternatively, is there a way to have `helm-do-grep` run recursively?

In other words, does helm have any commands to show results from a _recursive_ grep-type of search _interactively_ (i.e. as I type)?
